### PR TITLE
FX: granting '/replication' user permission to read/write buckets and objects

### PR DESCRIPTION
**Pull request by nicolas2bert**
This PR grants '/replication' user permission to read and write any buckets and objects.

### CODE REVIEW

File: lib/auth/in_memory/builder.js [hunk 1]
```diff
@@ -4,23 +4,23 @@ const serviceAccountPrefix =
 /** build simple authdata with only one account
  * @param {string} accessKey - account's accessKey
  * @param {string} secretKey - account's secretKey
+ * @param {string} canonicalId - account's canonical id
  * @param {string} [serviceName] - service name to use to generate can id
  * @return {object} authdata - authdata with account's accessKey and secretKey
  */
-function buildAuthDataAccount(accessKey, secretKey, serviceName) {
+function buildAuthDataAccount(accessKey, secretKey, canonicalId, serviceName) {
```
Comment (RachedBenMustapha) [inline lib/auth/in_memory/builder.js]: Now that service account canonical ids contain the service type, we can remove the `serviceName` handling altogether.

Comment (nicolas2bert) [reply lib/auth/in_memory/builder.js]: I kept `serviceName` for legacy purpose. If the canonicalId is not specified `buildAuthDataAccount` create a canonical id as: `${serviceAccountPrefix}/${serviceName}`

**Comment by nicolas2bert** [comment]
@dora-korpar I added some tests :)

**Comment by RachedBenMustapha** [comment]
Merging into the feature branch

**Pull request merged**